### PR TITLE
add prerelease lifecycle tags to autoscaling

### PR DESCRIPTION
**Pull request by deads2k**
This adds the lifecycle annotation tags matching https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/1635-prevent-permabeta/README.md 

The clock is set for removal of beta levels in 1.25.

/kind cleanup
/priority important-soon
@DirectXMan12  @gliush @kubernetes/sig-api-machinery-misc 

```release-note
autoscaling/v2beta1 is deprecated in favor of autoscaling/v2beta2
```

### CODE REVIEW

File: staging/src/k8s.io/api/autoscaling/v2beta2/types.go [hunk 1]
```diff
@@ -26,6 +26,9 @@ import (
 
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:prerelease-lifecycle-gen:introduced=1.12
+// +k8s:prerelease-lifecycle-gen:deprecated=1.19
+// +k8s:prerelease-lifecycle-gen:removed=1.22
```
Comment (liggitt) [inline staging/src/k8s.io/api/autoscaling/v2beta2/types.go:31]: deprecated 1.22, removed 1.25, per https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/1635-prevent-permabeta#impacted-apis

File: staging/src/k8s.io/api/autoscaling/v2beta2/types.go [hunk 2]
```diff
@@ -467,6 +470,9 @@ type MetricValueStatus struct {
 }
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:prerelease-lifecycle-gen:introduced=1.12
+// +k8s:prerelease-lifecycle-gen:deprecated=1.19
+// +k8s:prerelease-lifecycle-gen:removed=1.22
```
Comment (liggitt) [inline staging/src/k8s.io/api/autoscaling/v2beta2/types.go:475]: deprecated 1.22, removed 1.25, per https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/1635-prevent-permabeta#impacted-apis

**Comment by deads2k** [comment]
pushed

**Comment by liggitt** [comment]
/kind deprecation

needs release note for v2beta1 bits

**Review by mwielgus — APPROVED** [review]
/lgtm
/approve

**Pull request merged**